src/core/videosource.cpp

```diff
@@ -341,6 +341,8 @@ void FFMS_VideoSource::ReAdjustOutputFormat(AVFrame *Frame) {
     BCSType InputType = GuessCSType(InputFormat);
     BCSType OutputType = GuessCSType(OutputFormat);
 
+    OutputColorSpaceSet = (InputType != OutputType);
```

Comment:
Is this correct? Comparing pixel formats to determine if the color space and range were set doesn't seem valid (although it'll happen to work for the usual case of asking for RGB frames from YUV video).

Reply:
Maybe `Set` was a poor naming choice. The variable is meant to signify that no output range was chosen by FFMS2 (i.e. `SetOutputFormatV` wasn't called by the API user). I meant it to work in a similar way to the other properties that are initialized to `-1` below, and checked (`>= 0`) before assignment in `OutputFrame`.

Am I perhaps missing some understanding of the code?

Reply:
I think `SetOutputFormat()` should just be setting these fields to `true` (and `ResetOutputFormat()` needs to set them to `false`). It doesn't need to be conditionally set depending on whether or not the current frame's format happens to match the specified output format, because when they do it doesn't matter which field is read.

Reply:
Updated and tested with this change. Changes are signalled property.